### PR TITLE
[platform] add platform check for core configuration option

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: examples/platforms/nrf52840/Makefile.am [hunk 1]
```diff
@@ -86,6 +86,7 @@ PLATFORM_COMMON_SOURCES
     logging.c                                                                                                \
     misc.c                                                                                                   \
     openthread-core-nrf52840-config.h                                                                        \
+    openthread-core-nrf52840-config-platform-check.h                                                         \
```
Comment (LuDuda) [inline examples/platforms/nrf52840/Makefile.am]: nit: maybe just openthread-core-nrf52840-config-check.h?

File: examples/platforms/nrf52840/openthread-core-nrf52840-config-platform-check.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OPENTHREAD_PLATFORM_NRF52840_CORE_CONFIG_CHECK_H_
```
Comment (LuDuda) [inline examples/platforms/nrf52840/openthread-core-nrf52840-config-platform-check.h]: nit; align define with the file name

File: examples/platforms/gp712/Makefile.am [hunk 3]
```diff
@@ -47,6 +47,7 @@ PLATFORM_SOURCES                          = \
     logging.c                               \
     misc.c                                  \
     openthread-core-gp712-config.h          \
+    openthread-core-gp712-config-platform-check.h \
```
Comment (LuDuda) [inline examples/platforms/gp712/Makefile.am]: nit: alignment

File: examples/platforms/nrf52840/Makefile.am (hunk 1 above)
Comment (zhanglongxia) [reply examples/platforms/nrf52840/Makefile.am]: updated.

File: examples/platforms/gp712/Makefile.am (hunk 3 above)
Comment (zhanglongxia) [reply examples/platforms/gp712/Makefile.am]: Updated.

File: examples/platforms/efr32mg21/openthread-core-efr32-config-check.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OPENTHREAD_CORE_EFR32_CONFIG_CHECK_H_
+#define OPENTHREAD_CORE_EFR32_CONFIG_CHECK_H_
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+#error "Platform efr32 doesn't support configuration option: OPENTHREAD_CONFIG_ENABLE_TIME_SYNC"
```
Comment (jwhui) [inline examples/platforms/efr32mg21/openthread-core-efr32-config-check.h]: nit: "efr32" -> "efr32mg21"

File: examples/platforms/efr32mg21/openthread-core-efr32-config-check.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT
+#error "Platform efr32 doesn't support configuration option: OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT"
```
Comment (jwhui) [inline examples/platforms/efr32mg21/openthread-core-efr32-config-check.h]: nit: "efr32" -> "efr32mg21"

File: examples/platforms/efr32mg12/openthread-core-efr32-config-check.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OPENTHREAD_CORE_EFR32_CONFIG_CHECK_H_
+#define OPENTHREAD_CORE_EFR32_CONFIG_CHECK_H_
+
+#include "board_config.h"
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+#error "Platform efr32 doesn't support configuration option: OPENTHREAD_CONFIG_ENABLE_TIME_SYNC"
```
Comment (jwhui) [inline examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: nit: "efr32" -> "efr32mg12"

File: examples/platforms/efr32mg12/openthread-core-efr32-config-check.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+#endif
+
+#ifndef RADIO_CONFIG_915MHZ_OQPSK_SUPPORT
+#if OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT
+#error "Platform efr32 doesn't support configuration option: OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT"
```
Comment (jwhui) [inline examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: nit: "efr32" -> "efr32mg12"

Comment (jwhui) [inline examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: Suggest updating this to something like "Platform efr32mg12 not configured to support OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT."

File: examples/Makefile-cc1352 [hunk 8]
```diff
@@ -68,10 +68,9 @@ CC1352_MBEDTLS_CPPFLAGS                                    = \
     -I$(PWD)/third_party/mbedtls/repo/include                \
     $(NULL)
 
-CC1352_CONFIG_FILE_CPPFLAGS                                                   = \
-    -DOPENTHREAD_PROJECT_CORE_CONFIG_FILE='\"openthread-core-cc1352-config.h\"' \
-    -I$(PWD)/examples/platforms/cc1352/                                         \
-    $(NULL)
+CC1352_CONFIG_FILE_CPPFLAGS  = -DOPENTHREAD_PROJECT_CORE_CONFIG_FILE='\"openthread-core-cc1352-config.h\"'
+CC1352_CONFIG_FILE_CPPFLAGS += -DOPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE='\"openthread-core-cc1352-config-check.h\"'
+CC1352_CONFIG_FILE_CPPFLAGS += -I$(PWD)/examples/platforms/cc1352/
```
Comment (jwhui) [inline examples/Makefile-cc1352]: I see some variations in how these config file related CPPFLAGS are defined across the various platform makefiles. Can we try to make it consistent?

File: examples/platforms/efr32mg21/openthread-core-efr32-config-check.h (hunk 4 above)
Comment (zhanglongxia) [reply examples/platforms/efr32mg21/openthread-core-efr32-config-check.h]: Updated

File: examples/platforms/efr32mg12/openthread-core-efr32-config-check.h (hunk 6 above)
Comment (zhanglongxia) [reply examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: Updated

File: examples/platforms/efr32mg12/openthread-core-efr32-config-check.h (hunk 7 above)
Comment (zhanglongxia) [reply examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: Updated

Comment (zhanglongxia) [reply examples/platforms/efr32mg12/openthread-core-efr32-config-check.h]: Updated

File: examples/Makefile-cc1352 (hunk 8 above)
Comment (zhanglongxia) [reply examples/Makefile-cc1352]: Updated

**Pull request merged**